src/org/exist/collections/Collection.java

```diff
@@ -2305,4 +2303,32 @@ public QNameRangeIndexSpec getIndexByQNameConfiguration(final DBBroker broker, f
         final IndexSpec idxSpec = getIndexConfiguration(broker);
         return (idxSpec == null) ? null : idxSpec.getIndexByQName(qname);
     }
+
+    @Override
+    public ResourceMetadata getMetadata() {
+        return new CollectionMetadata();
```

Comment:
Yes what you have currently is a light operation, but I don't understand why you want to implement it differently to DocumentMetadata.